src/g_hslider.c

```diff
@@ -396,27 +392,22 @@ static void hslider_dialog(t_hslider *x, t_symbol *s, int argc, t_atom *argv)
 
 static void hslider_motion(t_hslider *x, t_floatarg dx, t_floatarg dy)
 {
-    int old = x->x_val;
+    int old = x->x_pos;
 
     if(x->x_gui.x_fsf.x_finemoved)
-        x->x_pos += (int)dx;
+        x->x_pos += (int)(dx);
     else
-        x->x_pos += 100 * (int)dx;
-    x->x_val = x->x_pos;
-    if(x->x_val > (100*x->x_gui.x_w - 100))
+        x->x_pos += (int)(100 * dx / IEMGUI_ZOOM(x));
+    if(x->x_pos > (100 * (x->x_gui.x_w / IEMGUI_ZOOM(x)) - 100))
```

Comment:
that looks bogus. why is the position not scaled by the zoom-factor when we <kbd>Shift</kbd>-drag?

actually i think that the entire `if`/`else` clause should be:

~~~C
x->x_pos += (int)(((x->x_gui.x_fsf.x_finemoved)?100.:1.) * dx / IEMGUI_ZOOM(x));
~~~

Reply:
> why is the position not scaled by the zoom-factor when we Shift-drag?

because half-pixel mouse steps would be rounded to 0, so you couldn't move the slider, excecpting if you moved the mouse quickly (so you have more than 2-pixels steps).

Reply:
hmm.
if you make `x_pos` a `t_float` then that wouldn't e an issue.

otoh, i now see that taking IEMGUI_ZOOM into account, will give finer precision of the slider when zoomed in.
i'm not sure whether this is actually desired.

e.g. a default slider will have integer increments. with your patch applied, it will suddenly have `0.5`-step increment in zoomed-mode which might make for some surprise.

Reply:
> make `x_pos` a `t_float`

yes, that's what I did with the [knob].
but it has some float-rounding side effects when you want to have integer increments, and anyway it wouldn't remove the effect you see (finer precision when zoomed).

I would see a way to solve that: we could keep track of the actual pointer position; this would require adding some local global variables. So we could [div] the movement by the zoom factor before applying the motion method.

But I'm not sure it's worth it? 
My first attempt hadn't that drawback. Too bad I lost it with force-pushing ;-) (maybe I should never do that)

Reply:
OK, please excuse my raving, the solution was finally rather simple, see https://github.com/pure-data/pure-data/pull/933/commits/6873180b88d3247a786865510a0850babf306350.

I think the behavior is now the same than before.